forms-flow-components/src/customHooks/useDownloadFile.ts

```diff
@@ -0,0 +1,45 @@
+import { useRef, useState, MutableRefObject } from "react";
+
+interface UseDownloadFileParams {
+  apiDefinition: () => Promise<{ data: any }>;
+  preDownloading: () => void;
+  postDownloading: () => void;
+  onError: (error?: any) => void;
+  getFileName: () => string;
+}
+
+interface UseDownloadFileReturn {
+  download: () => Promise<void>;
+  ref: MutableRefObject<HTMLAnchorElement | null>;
+  url: string | undefined;
+  name: string | undefined;
+}
+
+export const useDownloadFile = ({
+  apiDefinition,
+  preDownloading,
+  postDownloading,
+  onError,
+  getFileName,
+}: UseDownloadFileParams): UseDownloadFileReturn => {
+  const ref = useRef<HTMLAnchorElement | null>(null);
+  const [url, setFileUrl] = useState<string | undefined>();
```

Comment:
Articulate the useState in the proper way either like 
const [url, setUrl] = useState<string | undefined>();
or
const [fileUrl, setFileUrl] = useState<string | undefined>();